compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/NameCollision.hs

```diff
@@ -147,6 +147,8 @@ checkDataType moduleName DefDataType{..} =
             forM_ constrs $ \vconName -> do
                 checkName (NEnumCon moduleName dataTypeCon vconName)
 
+        DataSynonym _ -> return ()
```

Comment:
This should be part of the name collision check, or track it with a TODO :-)

```suggestion
        DataSynonym _ ->
            checkName (NSynonymType moduleName dataTypeCon)
```

and respective entries for `NSynonymType` in other functions.

Reply:
Ah. yes. I missed this.
I mistakenly thought it was only checking for collision ins the data-constructors.
But of course it also checks `dataTypeCon`, the name of the type.

Reply:
fixed